templates/gatsby-drupal/files/gatsby/start.sh

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+# Environment-specific configuration for Gatsby.
+if [ "$PLATFORM_BRANCH" = master ]; then
+   sleep infinity
+else
+   npm run develop
+fi
```

Comment:
Ah, that's why the main build only runs on master.  This is where inlining both scripts would make it easier to understand, since you can see them next to each other.  And inline docs can explain why it's done that way.